File: 1755-defuse-the-bomb/defuse-the-bomb.py
class Solution:
    def decrypt(self, code: List[int], k: int) -> List[int]:
        n = len(code)
        result = [0] * n

        for i in range(n):
            if(k >= 0):
                for j in range(1,k+1):
                    result[i] += code[(i+j)%n]
            else:
                for j in range(1, -k + 1):
                    result[i] += code[(i-j+n)%n]
        
        return result
        

File: 1755-defuse-the-bomb/defuse-the-bomb.cpp
class Solution {
public:
    vector<int> decrypt(vector<int>& code, int k) {
        vector<int> sum_arr(code.size());
        for(int i=0; i<code.size(); i++){
            if(k>=0){
                for(int j = 1; j<=k; j++){
                    sum_arr[i] += code[(i + j)%code.size()];
                }
            }else{
                for(int j = 1; j<=-k; j++){
                    sum_arr[i] += code[(i - j+ code.size())%code.size()];
                }
            }
            cout << "=" << sum_arr[i] << endl;

        }

        return sum_arr;
    }
};